src/figuresClasses.ts

```diff
@@ -2,10 +1,103 @@
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Triangle sides must be positive numbers');
+    }
+
+    const [side1, side2, side3] = [a, b, c].sort((x, y) => x - y);
+
+    if (side3 >= side1 + side2) {
+      throw new Error(`Sides ${a}, ${b} and ${c} can't form a triangle`);
+    }
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: 'circle';
 
-export class Triangle implements Figure {}
+  color: 'red' | 'green' | 'blue';
 
-export class Circle implements Figure {}
+  radius: number;
+
+  constructor(color: 'red' | 'green' | 'blue', radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be a positive number');
+    }
+    this.shape = 'circle';
+    this.color = color;
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    const area = Math.PI * Math.pow(this.radius, 2);
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: 'rectangle';
```

Comment:
The `shape` property should be declared as a constant string literal type: `shape: 'rectangle' = 'rectangle';` to ensure type safety and consistency with the `Figure` interface.